File: app/src/main/java/me/magnum/melonds/ui/input/FrontendInputHandler.java
package me.magnum.melonds.ui.input;

import me.magnum.melonds.model.Input;
import me.magnum.melonds.model.Point;

public abstract class FrontendInputHandler implements IInputListener {
    @Override
    public void onKeyPress(Input key) {
        switch (key) {
            case PAUSE:
                onPausePressed();
                break;
            case FAST_FORWARD:
                onFastForwardPressed();
                break;
        }
    }

    @Override
    public void onKeyReleased(Input key) {

    }

    @Override
    public void onTouch(Point point) {

    }

    public abstract void onPausePressed();
    public abstract void onFastForwardPressed();
}

File: app/src/main/java/me/magnum/melonds/ui/input/FrontendInputHandler.kt
package me.magnum.melonds.ui.input

import me.magnum.melonds.domain.model.Input
import me.magnum.melonds.domain.model.Point

abstract class FrontendInputHandler : IInputListener {
    override fun onKeyPress(key: Input) {
        when (key) {
            Input.PAUSE -> onPausePressed()
            Input.FAST_FORWARD -> onFastForwardPressed()
        }
    }

    override fun onKeyReleased(key: Input) {
    }

    override fun onTouch(point: Point) {
    }

    abstract fun onPausePressed()
    abstract fun onFastForwardPressed()
}